### PR TITLE
erik anderson code review submission

### PR DESCRIPTION
**Pull request by eriktuck1**
# Pull Request Template

### Reproducibility
- [ ] Are the data downloaded in the code
- [ ] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [ ] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [ ] Does the notebook run from start to finish?

### Functions
- [ ] Do functions follow PEP 8 format conventions?
- [ ] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [ ] Are function names expressive (the name describes what the function does)?
- [ ] Are functions easy to understand and read?
- [ ] How many tasks does each function do? (ideally a function does one thing well).

### Package imports
- [ ] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [ ] Are variable names throughout the code, expressive?
- [ ] Suggest changes if not, highlight what was done well
- [ ] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [ ] Are segments of code repeated in the file or is the code DRY?
- [ ] Are loops used to optimize DRY code?
- [ ] Are functions used to optimize DRY code?
- [ ] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [ ] Highlight any novel approaches to completing the assignment.

@eculler

### CODE REVIEW

File: anderson_erik_lidar_analysis.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+import os
+import pathlib
+
+import earthpy as et
+import matplotlib.pyplot as plt
+import pandas as pd
+import geopandas as gpd
+import rasterstats as rs
+import xarray as xr
```
Comment (StepanBryleev) [inline anderson_erik_lidar_analysis.py:19]: It seems that rioxarray library is not needed for this task and not used.

Comment (StepanBryleev) [inline anderson_erik_lidar_analysis.py:19]: It seems that xarray library is not needed for this task and not used.

File: anderson_erik_lidar_analysis.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+import rioxarray as rxr
+import seaborn as sns
+
+et.data.get_data('spatial-vector-lidar')
```
Comment (StepanBryleev) [inline anderson_erik_lidar_analysis.py:23]: You could add more # comments to clarify your set up.

File: anderson_erik_lidar_analysis.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+      chm_stats_gdf
+      .merge(insitu_height_df, left_on=id_col, right_index=True)
+  )
+  return stat_df
+
+
+# In[19]:
+
+
+def plot_comparison(df):
```
Comment (StepanBryleev) [inline anderson_erik_lidar_analysis.py:222]: Docstring with parameters and returns is missing

File: anderson_erik_lidar_analysis.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    ax2.set(xlim=(0, 30), ylim=(0, 30), aspect='equal')
+    ax2.set_title('Maximum')
+    sns.regplot(x='lidar_max', y='insitu_max', 
+                data=df,
+                color='purple',
+                ax=ax2)
+    ax2.plot((0, 1), (0, 1), transform=ax2.transAxes, ls='--', c='k')
+    plt.suptitle('Comparison of LiDAR and insitu height measurements at the \n'
+                'NEON San Joaquin Experimental Range (SJER) field site')
+    plt.tight_layout()
```
Comment (StepanBryleev) [inline anderson_erik_lidar_analysis.py:244]: You could add more # comments to clarify your plot settings

File: anderson_erik_lidar_analysis.py [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+plot_comparison(soap_df)
+
+
+# In[21]:
+
+
+# Calculate and plot SJER site
+sjer_df = calc_height_stats(sjer_plots_path, sjer_chm_path, sjer_insitu_path)
+plot_comparison(sjer_df)
+
```
Comment (StepanBryleev) [inline anderson_erik_lidar_analysis.py:262]: Good job @eriktuck  !

**Review by StepanBryleev — COMMENTED** [review]
# Pull Request Template

## Review Checklist

### Reproducibility
- [x] Are the data downloaded in the code
- [x] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [ ] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [x] Does the notebook run from start to finish?

### PEP 8 standards & Code Readability

### Functions
- [x] Do functions follow PEP 8 format conventions?
- [ ] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [x] Are function names expressive (the name describes what the function does)?
- [x] Are functions easy to understand and read?
- [x] How many tasks does each function do? (ideally a function does one thing well).

### Package imports
- [ ] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [x] Are variable names throughout the code, expressive?
- [x] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [x] Is the code DRY?
- [ ] Are loops used to optimize DRY code?
- [x] Are functions used to optimize DRY code?
- [x] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [ ] Highlight any novel approaches to completing the assignment.

File: anderson_erik_lidar_analysis.py (hunk 1 above)
Comment (eriktuck1) [reply anderson_erik_lidar_analysis.py:19]: thanks for calling that out, definitely copy/pasta on my end

Comment (eriktuck1) [reply anderson_erik_lidar_analysis.py:19]: copy/pasta again!

File: anderson_erik_lidar_analysis.py (hunk 2 above)
Comment (eriktuck1) [reply anderson_erik_lidar_analysis.py:23]: agreed, thanks for the recommendation

File: anderson_erik_lidar_analysis.py (hunk 3 above)
Comment (eriktuck1) [reply anderson_erik_lidar_analysis.py:222]: yep, definitely missed that one

File: anderson_erik_lidar_analysis.py (hunk 4 above)
Comment (eriktuck1) [reply anderson_erik_lidar_analysis.py:244]: will do, thanks!

**Comment by eculler** [comment]
@eriktuck please review https://github.com/earthlab-education/ea-lidar-uncertainty-review/pull/59 (looks like you already did - thanks!)

**Comment by eculler** [comment]
Nice work, Erik.

Just a small quibble - functions starting with underscores are usually reserved for internal methods (i.e. methods that are not meant for user access in a class)

**Pull request closed without merging**